projects/concurrency/cache.go

```diff
@@ -0,0 +1,124 @@
+package concurrency
+
+import (
+	"math"
+	"time"
+)
+
+type Statistics struct {
+	hitRate             int
```

Comment:
This isn't wrong, but I'd generally expect this to be a `float64` rather than an `int` - there's an interesting question as to how much precision we find valuable here, but rates generally are decimals.